src/vhost_user/message.rs

```diff
@@ -459,6 +479,49 @@ impl VhostUserMsgValidator for VhostUserMemoryRegion {
 /// Payload of the VhostUserMemory message.
 pub type VhostUserMemoryPayload = Vec<VhostUserMemoryRegion>;
 
+/// Single memory region descriptor as payload for ADD_MEM_REG and REM_MEM_REG
+/// requests.
+#[repr(packed)]
+#[derive(Default, Clone, Copy)]
+pub struct VhostUserSingleMemoryRegion {
+    /// Padding for correct alignment
+    pub padding: u64,
```

Comment:
Should it be private? No one should access this field.
BTW, the padding field really seems odd:(

Reply:
Oh yes it should be private.
And yes I agree the padding looks so weird :-/

Reply:
Out of curiosity, I took a look at the relevant code in QEMU, and I still don't understand why we need that padding. Relevant commits in QEMU:

```
f1aeb14b0809e313c74244d838645ed25e85ea63 Transmit vhost-user memory regions individually
3009edff8192991293fe9e2b50b0d90db83c4a89 vhost-user: fix VHOST_USER_ADD/REM_MEM_REG truncation
```

Reply:
Well I relied on the spec here: https://qemu.readthedocs.io/en/latest/interop/vhost-user.html#single-memory-region-description

Reply:
@sboeuf Yes, your implementation is correct, I was just wondering why we needed that in the first place, when it was added to QEMU.